modules/es-extensions/store/extensions/app/greg_swagger/pages/index.jag

```diff
@@ -0,0 +1,200 @@
+<%
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+require('/modules/store.js').exec(function (ctx) {
+    var assetManager = function (session, type) {
+        var options = {'type': type};
+        var tenantResources = tenantApi.createTenantAwareAssetResources(session, options);
+        return tenantResources.am;
+    };
+
+    var tenantApi = require('/modules/tenant-api.js').api;
+    var server = require('store').server;
+    var carbon = require('carbon');
+    var config = require('/config/store.js').config();
+    var permissionAPI = require('rxt').permissions;
+
+    var ByteArrayInputStream = Packages.java.io.ByteArrayInputStream,
+            commonConstants = Packages.org.wso2.carbon.registry.extensions.utils.CommonConstants;
+    var jsonValue;
+    var path = ctx.params.path;
+    var tenantId = ctx.tenant.tenantId;
+
+    if (Boolean(path)) {
+        var am = assetManager(session, type);
+        var registry = am.registry.registry;
+
+        var resource;
+        if (registry.resourceExists(path)) {
+            resource = registry.get(path);
+        }
+
+        var content = resource.getContent();
+        var value = '' + new Stream(new ByteArrayInputStream(content));
+        jsonValue = parse(value);
+        var url = config.server.https;
+
+        if (commonConstants.SWAGGER_MEDIA_TYPE.equals(resource.getMediaType())) {
+            /*log.info("wrong content type");
+             response.status = 400;
+             print("swagger UI generation is only avaialable for swagger artifacts," +
+             " Please specify a registry path of a swagger artifact!");
+             return;*/
```

Comment:
Thanks, I have fixed it in [this commit](https://github.com/wso2/product-greg/pull/854/commits/93508144b9b01b4b424867b5632941c2166bf0e1)